server/src/main/java/org/opensearch/cluster/routing/WeightedRouting.java

```diff
@@ -47,6 +47,7 @@ public WeightedRouting(StreamInput in) throws IOException {
     }
 
     public boolean isSet() {
+        if (this.attributeName == null || this.weights == null) return false;
         return (!this.attributeName.isEmpty() && !this.weights.isEmpty());
```

Comment:
we can combine this in one line  : `this.weights != null && this.weights.isEmpty()`

Reply:
Updated